src/synthesized_insight/metrics/base.py

```diff
@@ -83,6 +84,81 @@ def __call__(self, sr_a: pd.Series, sr_b: pd.Series) -> Tuple[Union[int, float,
         return self._compute_test(sr_a, sr_b)
 
 
+class TwoColumnMap(_Metric):
+    """Compares columns with the same name from two given dataframes and return a DataFrame
+    with index as the column name and the columns as metric_value and metric_pval(if applicable)"""
+
+    def __init__(self, metric: Union[TwoColumnMetric, TwoColumnTest]):
+        self.metric = metric
+        self.name = f'{metric.name}_map'
+        super().__init__()
+
+    def __call__(self, df_old: pd.DataFrame, df_new: pd.DataFrame) -> Union[pd.DataFrame, None]:
+        if df_old is None or df_new is None:
+            return None
+
+        columns_map = {col: self.metric(df_old[col], df_new[col]) for col in df_old.columns}
+
+        result = pd.DataFrame(
+            data=columns_map.values(),
+            index=df_old.columns,
+            columns=['metric_val', 'metric_pval'] if isinstance(self.metric, TwoColumnTest) else ['metric_val']
+        )
+
+        return result
+
+
+class CorrMatrix(_Metric):
```

Comment:
Like the comment above with '_Series', there should be a base class _Matrix, which returns a dataframe on the __call__ function

Reply:
CorrMatrix acts on a single dataframe and returns a correlation matrix dataframe, hence changed the base class from _Metric to DataFrameMatrix